### DISCUSSION TITLE
Canonical solution to negative cycles due to rounding errors in exchange rate graphs?

### DISCUSSION THREAD

**Discussion by ff781** [networkx/networkx#7584] [Q&A] [answered]
What is the canonical solution to barely negative cycles due to rounding errors in exchange rate graphs, where often there is a trivial 0 cost cycle?

Between two nodes where there is bidirectional exchange, assuming fully frictionless, the log exchange range weights are the additive inverse in the flipped direction. But due to rounding errors, this might result in a negative cycle, which fails e.g. bellman-ford.

**Comment by dschult** [comment, marked as answer]
If possible, convert your edge weights to integers by choosing some level of precision. Then you won't get round-off.
e.g. with weights 1.2345678910, 2.345678991011 choose 6 digits to convert them to 123456, 234567 using something like:
```py
nx.set_edge_attributes(G, {(u, v): int(wt * 100000) for u, v, wt in G.edges(data='weight'), 'rounded_weight')
```
You can select as many digits as you like.
